src/components/TextInput/BaseTextInput.js

```diff
@@ -59,7 +59,8 @@ class BaseTextInput extends Component {
 
     componentDidUpdate() {
         // Activate or deactivate the label when value is changed programmatically from outside
-        if (this.value === this.props.value) {
+        // Only update when value prop is provided
+        if (this.props.value === undefined || this.value === this.props.value) {
```

Comment:
NAB `_.isUndefined(this.props.value)`

Reply:
Yeah, I will do that in https://github.com/Expensify/App/issues/6584 during refactoring.